test/functional/apps/discover/group1/_discover.ts

```diff
@@ -170,10 +170,24 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       });
 
       it('should show matches when time range is expanded', async () => {
-        await retry.waitFor('view all matches to load', async () => {
-          await PageObjects.discover.expandTimeRangeAsSuggestedInNoResultsMessage();
+        await PageObjects.discover.expandTimeRangeAsSuggestedInNoResultsMessage();
+        await retry.waitForWithTimeout('view all matches to load', 60000, async () => {
           await PageObjects.discover.waitUntilSearchingHasFinished();
-          return !(await testSubjects.exists('discoverNoResultsViewAllMatches'));
+          if (!(await testSubjects.exists('discoverNoResultsViewAllMatches'))) {
+            return true;
+          }
+          try {
+            // If the button still exists, the click didn't work, so try again.
```

Comment:
I have not observed cases where a click did not work. Usually the UI needs more time to update.

This approach seems to be passing too https://github.com/elastic/kibana/pull/177682

Reply:
Okay, looking at latest failures, I think none of us PRs is actually addressing them

```
timed out waiting for the button before pressing it
```

Currently, there is no button shown yet 🤷‍♀️

Reply:
Nvm. I am okay with merging the PR although would be better to move the logic inside `expandTimeRangeAsSuggestedInNoResultsMessage` to avoid duplication and consider the fact that there is an request happening after pressing the button. No need to press it again if it's in loading state.

Reply:
That only started occurring after my last attempt to fix it, and I'm pretty certain that's because my last PR introduced separate flakiness here: https://github.com/elastic/kibana/blob/5a9cd830271491a2e7393686b5fd0a22aeb09e9e/x-pack/test_serverless/functional/test_suites/common/discover/group1/_discover.ts#L179-L183

I suspect this is what's happening with the new failures:
- Button is clicked on line 180, and loading starts successfully
- Loading is still in progress and the button still exists when checked on line 182, so it retries
- By the time the retry starts and line 180 runs again, the loading has completed and the button has disappeared, so the attempt to re-click the button fails

That's why I went with the approach I did that attempts to re-click the button, but continues if it fails since that likely means the loading actually succeeded and the button was hidden. This should handle both the cases where the initial click was successful, and also when the initial click fails so a re-click is necessary.

Reply:
Oops I had the same browser tab open when I came back to this and it only updated with your first comment, so I was replying to that. I'm seeing your second comment now, and sounds good, I'll see what I can do 👍

Reply:
I moved the logic to `expandTimeRangeAsSuggestedInNoResultsMessage` to avoid duplicating it and added a check to skip re-click attempts when the button is disabled: https://github.com/elastic/kibana/pull/177530/commits/05df74d5a7ea9fc1af7d910165c97aef89dac811. It also passed another 95x flaky test run, so hopefully it fixes the issue in `main` 🤞